c7n/resources/elasticsearch.py

```diff
@@ -115,6 +117,99 @@ class KmsFilter(KmsRelatedFilter):
     RelatedIdsExpression = 'EncryptionAtRestOptions.KmsKeyId'
 
 
+@ElasticSearchDomain.filter_registry.register('cross-account')
+class ElasticSearchCrossAccountAccessFilter(CrossAccountAccessFilter):
+    """
+    Filter to return all elasticsearch domains with cross account access permissions
+
+    :example:
+
+    .. code-block:: yaml
+
+        policies:
+          - name: check-elasticsearch-cross-account
+            resource: aws.elasticsearch
+            filters:
+              - type: cross-account
+    """
+    policy_attribute = 'c7n:Policy'
+    permissions = ('es:DescribeElasticsearchDomainConfig',)
+
+    def process(self, resources, event=None):
+        client = local_session(self.manager.session_factory).client('es')
+        for r in resources:
+            result = self.manager.retry(
+                client.describe_elasticsearch_domain_config,
+                DomainName=r['DomainName'],
+                ignore_err_codes=('ResourceNotFoundException',))
+            r[self.policy_attribute] = json.loads(
```

Comment:
if resource is not found, result will be none, this should be checked for, and if its the case continue in the loop